### DISCUSSION TITLE
Deleting stored VCs

### DISCUSSION THREAD

**Discussion by atz3n** [decentralized-identity/veramo#629] [Q&A] [answered]
Hello again,
I just wanted to know what the best method would be to delete stored VCs. Can't find an interface for that. Maybe I'm missing something.

Cheers

**Comment by mirceanis** [comment, marked as answer]
You are not missing anything.
The interface for deleting VCs was never built.

**Reply by atz3n** [reply]
Ok, thanks. Is there a plan to add such an interface? I would do it on my own but I'm unfortunately a noob in terms of databases and TypeORM :). I've already tried to delete a VC with the following code:
```ts
try {
    const credentialEntity = await (await this.dbConnection)
        .getRepository(Credential)
        .findOneOrFail(args.hash)
        
    await (await this.dbConnection)
        .getRepository(Credential)
        .remove(credentialEntity)
} catch (e) {
    throw Error('Verifiable credential removed') 
}
```

but I get the following error:
```

QueryFailedError: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed
    at new QueryFailedError (.../src/error/QueryFailedError.ts:11:9)
    at Statement.handler (.../src/driver/sqlite/SqliteQueryRunner.ts:79:26)
    at Statement.replacement (.../node_modules/sqlite3/lib/trace.js:25:27) {
  errno: 19,
  code: 'SQLITE_CONSTRAINT',
  __augmented: true,
  query: 'DELETE FROM "credential" WHERE "hash" = ?',
  parameters: [
    '8f075afdde3714803627902a54efe5183becca55f1b170b50b226597da869f986b13fcf5a2f5801d429dba1c72b0051a18d5f52a872a2f73ca1b5986069a97c8'
  ]
}
```

after some googling I could figure out that this is because there are some references to other tables. Could you give me a hint on which tables are referenced here or what steps are needed to properly delete a VC?

**Reply by simonas-notcat** [reply]
Hey @atz3n, the foreign constraint is on the `Claim` entity. Here is a PR that adds this functionality: https://github.com/uport-project/veramo/pull/634

**Reply by atz3n** [reply]
😲 That was fast... Thanks a lot. Saved me a lot of time :)
